src/utils/searchAndSort.js

```diff
@@ -0,0 +1,177 @@
+export function searchList(type, value, searchInput, list) {
+	if (type === 'search') {
+		if (value !== '') {
+			list = list.filter(challenge => {
+				return challenge.title.toLowerCase().includes(value.toLowerCase());
+			});
+		}
+	} else {
+		if (searchInput !== '') {
+			list = list.filter(challenge => {
+				return challenge.title.toLowerCase().includes(searchInput.toLowerCase());
+			});
+		}
+	}
+	return list;
+}
```

Comment:
This could maybe be cleaned up a little with something like this:
```
if (type === 'search' && value !== '') {
	list = list.filter(challenge => {
		return challenge.title.toLowerCase().includes(value.toLowerCase());
	});
} 
	
if (searchInput !== '') {
	list = list.filter(challenge => {
		return challenge.title.toLowerCase().includes(searchInput.toLowerCase());
	});
}

return list;
}
```